### ISSUE TITLE
v0.11.1 + PR #782: Still issues with unit tests

### ISSUE THREAD

**Issue by wiene** [lark-parser/lark#792]
Even after applying the patch from PR #782 I am still having issues with the unit tests. I tried out the following three ways to run them:

1. `PYTHONWARNINGS=d python3.9 -m unittest discover -v tests`
2. `python3.9 setup.py test`
3. `python3.9 -m tests`

The outcome is:

* v0.11.1: Only option 3 runs without problems.
* v0.11.1 + PR #782: Option 2 and 3 run successfully. Option 1 still fails with:
```
======================================================================
ERROR: test_postlex_declare (test_parser.tests.test_parser.TestEarleyDynamic_complete)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/autopkgtest.GL4ca6/autopkgtest_tmp/tests/test_parser.py", line 1895, in test_postlex_declare
    tree = parser.parse(test_file)
  File "/tmp/autopkgtest.GL4ca6/autopkgtest_tmp/lark/lark.py", line 495, in parse
    return self.parser.parse(text, start=start)
  File "/tmp/autopkgtest.GL4ca6/autopkgtest_tmp/lark/parser_frontends.py", line 223, in parse
    return self._parse(start, text)
  File "/tmp/autopkgtest.GL4ca6/autopkgtest_tmp/lark/parser_frontends.py", line 72, in _parse
    return self.parser.parse(input, start, *args)
  File "/tmp/autopkgtest.GL4ca6/autopkgtest_tmp/lark/parsers/earley.py", line 294, in parse
    to_scan = self._parse(stream, columns, to_scan, start_symbol)
  File "/tmp/autopkgtest.GL4ca6/autopkgtest_tmp/lark/parsers/xearley.py", line 139, in _parse
    to_scan = scan(i, to_scan)
  File "/tmp/autopkgtest.GL4ca6/autopkgtest_tmp/lark/parsers/xearley.py", line 53, in scan
    m = match(item.expect, stream, i)
  File "/tmp/autopkgtest.GL4ca6/autopkgtest_tmp/lark/parser_frontends.py", line 202, in match
    return self.regexps[term.name].match(text, index)
KeyError: 'B'

======================================================================
ERROR: test_postlex_indenter (test_parser.tests.test_parser.TestEarleyDynamic_complete)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/autopkgtest.GL4ca6/autopkgtest_tmp/tests/test_parser.py", line 1918, in test_postlex_indenter
    parser.parse("a\n    b\n")
  File "/tmp/autopkgtest.GL4ca6/autopkgtest_tmp/lark/lark.py", line 495, in parse
    return self.parser.parse(text, start=start)
  File "/tmp/autopkgtest.GL4ca6/autopkgtest_tmp/lark/parser_frontends.py", line 223, in parse
    return self._parse(start, text)
  File "/tmp/autopkgtest.GL4ca6/autopkgtest_tmp/lark/parser_frontends.py", line 72, in _parse
    return self.parser.parse(input, start, *args)
  File "/tmp/autopkgtest.GL4ca6/autopkgtest_tmp/lark/parsers/earley.py", line 294, in parse
    to_scan = self._parse(stream, columns, to_scan, start_symbol)
  File "/tmp/autopkgtest.GL4ca6/autopkgtest_tmp/lark/parsers/xearley.py", line 139, in _parse
    to_scan = scan(i, to_scan)
  File "/tmp/autopkgtest.GL4ca6/autopkgtest_tmp/lark/parsers/xearley.py", line 53, in scan
    m = match(item.expect, stream, i)
  File "/tmp/autopkgtest.GL4ca6/autopkgtest_tmp/lark/parser_frontends.py", line 202, in match
    return self.regexps[term.name].match(text, index)
KeyError: 'INDENT'
```
@MegaIng: Any idea why option 1 still fails?

/cc @merkys

**Comment by MegaIng** [comment]
Can't reproduce. Are you using the most recent master? After all, #782 is already merged.

**Comment by MegaIng** [comment]
Ok, figured it out: The tests are simply broken. Don't randomly apply commits to past versions. Just use the most recent one. This is an impossible to fix. (I mean, it is fixed.)

**Comment by MegaIng** [comment]
Also, using `python3.9 -m unittest discover -v` works.

**Comment by MegaIng** [comment]
If you want to not use master, the parts that fixed this for the most recent master are in #776.

**Comment by wiene** [comment]
@MegaIng: Thanks for checking. I've just checked the master and can confirm that all three options work on the present master. Nevertheless I prefer using releases for inclusion in Debian rather than intermediate commits. Therefore my hope was that the problem in v0.11.1 could be fixed with a concise and dedicated patch. If this is not the case, I will probably wait for the next release. Thanks a lot for your feedback.

**Comment by MegaIng** [comment]
The problem is that a TestCase is run that is wrongly not run with the other methods. This TestCase has a bug that was fixed as part of #776. I am not sure why you can't use any of the other options to run tests, but removing the `('earley', 'dynamic_complete')` entry at the bottom of `test_parser.py` will also fix this.

**Comment by MegaIng** [comment]
But in the better way to fix this permanently is for you to change how the tests are run. We might at some point decide not to run an entire TestCase on purpose, which would cause similar problems again.
